### PR TITLE
Aa/nested divisions

### PR DESCRIPTION
**Pull request by adeola-ak**
going forward ill be more aware or limiting changes to the associated branch and commits. Sorry this PR is all over the place

### CODE REVIEW

File: .okta.env [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export OKTA_OAUTH2_ISSUER="https://dev-75062498.okta.com/oauth2/default"
+export OKTA_OAUTH2_CLIENT_ID="0oa3x0ow2tdEJYs9h5d7"
+export OKTA_OAUTH2_CLIENT_SECRET="Zpuy8HZ-EVkENpNb7YgpaDoY6d5a0UxHF-JQzboa"
```
Comment (courier-new) [inline .okta.env:3]: Uh oh, we definitely don't want these variables checked in! Might want to rotate that secret key while you're here! 😅

File: components/AllEmployees/AllEmployees.tsx [hunk 2]
```diff
@@ -5,69 +5,123 @@ import Nav from "react-bootstrap/Nav";
...
-    divisions.add(division);
-    if (count[division] === undefined) {
-      count[division] = 0;
+
+  allEmployees.forEach((employee) => {
+    const department = employee.department;
+    const division = employee.division;
+    if (!department) {
+      return;
+    }
```
Comment (courier-new) [inline components/AllEmployees/AllEmployees.tsx:23]: Could be helpful to leave a note explaining how the Bamboo API works here, something like "New hires who have not yet started will usually not have a department."

File: components/AllEmployees/AllEmployees.tsx [hunk 3]
```diff
@@ -5,69 +5,123 @@ import Nav from "react-bootstrap/Nav";
...
+                        >
+                          {
+                            <span className={styles.departmentCount}>
+                              {department}
+                            </span>
+                          }
+                          {
+                            // <span className={styles.departmentCount}>
+                            //   {count[department]}
+                            // </span>
```
Comment (courier-new) [inline components/AllEmployees/AllEmployees.tsx:89]: Can we remove this part now?

File: components/AllEmployees/AllEmployees.tsx [hunk 4]
```diff
@@ -5,69 +5,123 @@ import Nav from "react-bootstrap/Nav";
...
+                        eventKey={`event-${department}-${selectedDivision}`}
+                        className={styles.tabPane}
+                      >
+                        <div className={styles.tabPane}>
+                          {allEmployees
+                            .filter(
+                              (employee) =>
+                                employee.division === selectedDivision &&
+                                employee.department === selectedDepartment,
+                              // employee.status != "Inactive",
```
Comment (courier-new) [inline components/AllEmployees/AllEmployees.tsx:115]: Do we not need this bit anymore?

File: components/Carousel/Carousel.tsx [hunk 5]
```diff
@@ -1,47 +1,71 @@
...
   const leftClick = React.useCallback(() => {
-    let newActive = active;
-    newActive--;
-    setActive(newActive < 0 ? activeNewHires.length - 1 : newActive);
+    setLeftMostIndex(
+      leftMostIndex === 0 ? activeNewHires.length - 1 : leftMostIndex - 1,
+    );
+    setRightMostIndex(
+      rightMostIndex === 0 ? activeNewHires.length - 1 : rightMostIndex - 1,
+    );
```
Comment (courier-new) [inline components/Carousel/Carousel.tsx:51]: Were you able to work out the bug with the carousel yet? 🙂 No need to change anything here for now, just wanted to ask if you had worked it out!

File: components/Modal/Modal.tsx [hunk 6]
```diff
@@ -22,7 +22,7 @@ const Modal = ({ setModal, employee }) => {
           <div className={styles.modalRightPanel}>
             <div className={styles.employeeInfo}>
               <span className={styles.employeeInfoText}>
-                {employee.displayName}
+                {employee.preferredName}
```
Comment (courier-new) [inline components/Modal/Modal.tsx:25]: Nice!

**Review by courier-new — COMMENTED** [review]
Looking great! Glad to see how you worked out the Okta stuff. Had a couple small comments -- mostly I think we'll just want to remove the secrets from the `.env` file because, well, secrets! Shhhhh! 😂

File: components/Carousel/Carousel.tsx (hunk 5 above)
Comment (adeola-ak) [reply components/Carousel/Carousel.tsx:51]: no :(

File: .okta.env (hunk 1 above)
Comment (adeola-ak) [reply .okta.env:3]: omggggg smh.

**Pull request merged**